cli/lint-tree.js

```diff
@@ -26,20 +26,29 @@ export default class LintTree extends Filter {
 
   processString(content, relativePath) {
     let report = this.cli.executeOnText(content, path.join(this.rootDir, relativePath));
-    let result = report.results[0] || {};
-    let errors = result.messages || [];
-
-    errors = errors.filter((error) => !IGNORED_FILE_MESSAGE_REGEXP.test(error.message));
 
     if (this.generateTests) {
+      let result = report.results[0] || {};
+      let errors = result.messages || [];
+      errors = errors.filter((error) => !IGNORED_FILE_MESSAGE_REGEXP.test(error.message));
       return this.testGenerator(relativePath, errors);
     }
 
-    errors.forEach((error) => {
-      ui.warn(dedent`
-        ${ error.source }
-        ${ error.message } (${ error.ruleId })
-      `);
+    // Display to the console
+    report.results.forEach((fileResult) => {
```

Comment:
I _think_ report.results will only ever contain a single array element here, since we are running it against a single file. Hence the `let result = report.results[0]`.